allure-junit5/src/main/java/io/qameta/allure/junit5/AllureJunit5AnnotationProcessor.java

```diff
@@ -0,0 +1,115 @@
+package io.qameta.allure.junit5;
+
+import io.qameta.allure.Allure;
+import io.qameta.allure.AllureLifecycle;
+import io.qameta.allure.Epic;
+import io.qameta.allure.Feature;
+import io.qameta.allure.Owner;
+import io.qameta.allure.Severity;
+import io.qameta.allure.Story;
+import io.qameta.allure.model.Label;
+import io.qameta.allure.model.Link;
+import io.qameta.allure.util.ResultsUtils;
+import org.junit.jupiter.api.extension.BeforeTestExecutionCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.lang.annotation.Annotation;
+import java.lang.annotation.Repeatable;
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Allure Junit5 annotation processor.
+ */
+public class AllureJunit5AnnotationProcessor implements BeforeTestExecutionCallback {
+
+    private static AllureLifecycle lifecycle;
```

Comment:
why static? 

I guess because extension loaded by SPI. In that case I rather prefer disable SPI and provide extension via setting it directly to launcher

Reply:
It's for test purpose only. I didn't find how set up extension to launcher.